File: bestia-models/src/test/java/bestia/model/domain/PlayerItemTest.java
package bestia.model.domain;

import org.junit.Test;

public class PlayerItemTest {

	// TODO Equipment spezifische Tests.
	
	@Test(expected = IllegalArgumentException.class)
	public void negative_amount_test() {
		Item i = new Item();
		Account a = new Account();
		PlayerItem pi = new PlayerItem(i, a, 4);
		pi.setAmount(-1);
	}
	
	@Test(expected = IllegalArgumentException.class)
	public void zero_amount_test() {
		Item i = new Item();
		Account a = new Account();
		PlayerItem pi = new PlayerItem(i, a, 4);
		pi.setAmount(0);
	}
	
	@Test
	@SuppressWarnings("unused")
	public void instance_test() {
		Item i = new Item();
		Account a = new Account();
		PlayerItem pi = new PlayerItem(i, a, 2);
	}
	
	@Test(expected = IllegalArgumentException.class)
	@SuppressWarnings("unused")
	public void null_instance_1_test() {
		PlayerItem pi = new PlayerItem(null, null, 0);
	}
	
	@Test(expected = IllegalArgumentException.class)
	@SuppressWarnings("unused")
	public void null_instance_2_test() {
		Item i = new Item();
		PlayerItem pi = new PlayerItem(i, null, 0);
	}
	
	@Test(expected = IllegalArgumentException.class)
	@SuppressWarnings("unused")
	public void null_instance_3_test() {
		Item i = new Item();
		Account a = new Account();
		PlayerItem pi = new PlayerItem(i, a, 0);
	}
	
	@Test(expected = IllegalArgumentException.class)
	public void null_acc_test() {
		PlayerItem pi = new PlayerItem();
		pi.setAccount(null);
	}
	
	@Test(expected=IllegalArgumentException.class)
	public void null_item_test() {
		PlayerItem pi = new PlayerItem();
		pi.setItem(null);
	}
	
}

File: bestia-models/src/test/java/net/bestia/model/domain/PlayerItemTest.kt
package net.bestia.model.domain

import net.bestia.model.item.Item
import net.bestia.model.item.PlayerItem
import net.bestia.model.test.AccountFixture
import org.junit.Test

class PlayerItemTest {

  @Test(expected = IllegalArgumentException::class)
  fun negative_amount_test() {
    val i = Item()
    val a = AccountFixture.createAccount()
    val pi = PlayerItem(i, a, 4)
    pi.amount = -1
  }

  @Test(expected = IllegalArgumentException::class)
  fun zero_amount_test() {
    val i = Item()
    val a = AccountFixture.createAccount()
    val pi = PlayerItem(i, a, 4)
    pi.amount = 0
  }

  @Test
  fun instance_test() {
    val i = Item()
    val a = AccountFixture.createAccount()
    val pi = PlayerItem(i, a, 2)
  }

  @Test(expected = IllegalArgumentException::class)
  fun null_instance_1_test() {
    val pi = PlayerItem(null, null, 0)
  }

  @Test(expected = IllegalArgumentException::class)
  fun null_instance_2_test() {
    val i = Item()
    val pi = PlayerItem(i, null, 0)
  }

  @Test(expected = IllegalArgumentException::class)
  fun null_instance_3_test() {
    val i = Item()
    val a = AccountFixture.createAccount()
    val pi = PlayerItem(i, a, 0)
  }

  @Test(expected = IllegalArgumentException::class)
  fun null_acc_test() {
    val pi = PlayerItem()
    pi.account = null
  }

  @Test(expected = IllegalArgumentException::class)
  fun null_item_test() {
    val pi = PlayerItem()
    pi.item = null
  }

}
